pkg/controller/service/service_controller.go

```diff
@@ -334,9 +335,9 @@ func (s *ServiceController) createLoadBalancerIfNeeded(key string, service *v1.S
 }
 
 func (s *ServiceController) persistUpdate(service *v1.Service) error {
-	var err error
-	for i := 0; i < clientRetryCount; i++ {
-		_, err = s.kubeClient.CoreV1().Services(service.Namespace).UpdateStatus(service)
+	retryOptions := wait.Backoff{Duration: clientRetryInterval, Steps: clientRetryCount}
+	return retry.RetryOnConflict(retryOptions, func() error {
```

Comment:
I think we want to retry on any error, not just conflicts

Reply:
@justinsb 
Oh, sorry, I didn't notice it. If this is the case, there seems to be no public function in the client-go to meet the requirements

Reply:
And, from a closer look, the need to replace this code is not great, I think it is okay to stay the same.
WDYT？@justinsb